### ISSUE TITLE
Multirun doesn't select the right function for simulating the project

### ISSUE THREAD

**Issue by ire4ever1190** [automatarium/automatarium#273]
[Reproducible example](https://pastebin.com/raw/EZVSnCvz)

In the multirun for the PDA add strings (should be there if you load the JSON file)
- `AAA` (Should fail)
- `AAAA` (Should pass)

Building the same machine in JFLAP gives those results but Automatarium says they both pass.
Also weird, `AAAA` doesn't pass in normal run since it jumps to the final node too fast (this is a separate issue it seems)

More investigation finds that this line is the problem
https://github.com/automatarium/automatarium/blob/56feb729b3b8a33f6d48101cc0480f8849322dc9/frontend/src/components/Sidepanel/Panels/TestingLab/TestingLab.js#L355
Since it means it always simulates multirun as an FSA (hence it ignores the stack requirement).

This logic error seems to appear in a few places. We should probably make a function `simulateGraph` that properly chooses how to simulate the project instead of all the `if`'s that are currently being used

**Comment by tdib** [comment]
I wouldn't say it straight up doesn't work, I think more investigation is required into the specific issue but atm it seems like the testing lab trace does it correctly, and multi-run seems to ignore an empty stack (an entirely wrong string will still be rejected)

**Comment by ire4ever1190** [comment]
Yeah wrote the issue late and didn't investigate properly.

But think I've found the root issue, seems like easy enough to fix (fix will come soon, we're still getting all the initial documents finished before we start sprinting)
